### PR TITLE
MM-21632: fix toggling interactive dialog boolean

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
My fix for [MM-17519](https://mattermost.atlassian.net/browse/MM-17519) broke the /rendering/ of the boolean toggle, but not the underlying interactive dialog state (and the thrust of the original issue). I think I was too focussed on the encoding in the original issue to realize the UI had not reflected the change.

My original change was [reverted](https://github.com/mattermost/mattermost-mobile/pull/3803) in v1.27, 
but this ticket continues to capture the required changes to complete the effort.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-21632
https://mattermost.atlassian.net/browse/MM-21683

#### Checklist
- [x] Added or updated unit tests (required for all new features)

#### Device Information
This PR was tested on: iOS Simulator

### CODE REVIEW

File: app/screens/interactive_dialog/dialog_element.js [hunk 1]
```diff
@@ -158,7 +158,7 @@ export default class DialogElement extends PureComponent {
                 <BoolSetting
                     id={name}
                     label={displayName}
-                    value={value === 'true'}
+                    value={Boolean(value)}
```
Comment (saturninoabril) [inline app/screens/interactive_dialog/dialog_element.js]: Toggling works now but there's a discrepancy on initial render when the default value (which is of [type string](https://docs.mattermost.com/developer/interactive-dialogs.html#checkbox-element)) is `"False"` or `"false"`.
Please take a look again.

Comment (lieut-data) [reply app/screens/interactive_dialog/dialog_element.js]: Ah, that's a truly unfortunate requirement from the data model. I'll extend the demo plugin with default cases here, tweak, and test again.

Comment (lieut-data) [reply app/screens/interactive_dialog/dialog_element.js]: It looks like the webapp is broken here too in the same way. That, and we show the "required bit" for these fields, but it's not really clear what that means (required `true`?). I'll work towards patching all of it.

**Comment by jgilliam17** [comment]
@lieut-data 
Can you please add the test steps? I am not sure how to get the interactive dialog up to verify. Thanks.

**Comment by lieut-data** [comment]
Hey @jgilliam17! I've commented on the JIRA ticket re: the use of the demo plugin's `/dialog` command, but note that some changes to this PR are still pending given @saturninoabril's feedback above. There will be new PRs to the webapp and demo plugins as a consequence.

**Comment by lieut-data** [comment]
Opting to revert my original change, leaving the dissonance between `true` and `"true"` in place for now. Will continue this effort in the v1.28 timeframe.

**Comment by lieut-data** [comment]
@migbot, re-requesting review, given changes as discussed. Note the fixes for https://mattermost.atlassian.net/browse/MM-21683 are now included here to address @saturninoabril's feedback.

**Pull request merged**